### PR TITLE
Move BuildPath to com.google.cloud.tools.eclipse.appengine.libraries

### PR DESCRIPTION
**Pull request by elharo**
I'm moving BuildPath to com.google.cloud.tools.eclipse.appengine.libraries in preparation for using it after project creation to add a library to an existing project, not just when a new project is created. 

Pay special attention to how the context is created.

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/BuildPath.java [hunk 1]
```diff
@@ -40,9 +40,10 @@
 import com.google.cloud.tools.eclipse.appengine.libraries.LibraryClasspathContainerResolverJob;
 import com.google.cloud.tools.eclipse.appengine.libraries.model.Library;
 
-class BuildPath {
+public class BuildPath {
 
-  static void addLibraries(IProject project, List<Library> libraries, IProgressMonitor monitor)
+  public static void addLibraries(IProject project, List<Library> libraries, 
+      IProgressMonitor monitor, Class<?> classFromBundle)
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/BuildPath.java]: I think it's customary to have the monitor as a last parameter.

**Comment by briandealwis** [comment]
Oxygen failure from #1373.

File: plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/BuildPath.java [hunk 2]
```diff
@@ -73,12 +74,13 @@ static void addLibraries(IProject project, List<Library> libraries, IProgressMon
...
+    IEclipseContext context = EclipseContextFactory.getServiceContext(
+        FrameworkUtil.getBundle(classFromBundle).getBundleContext());
+    
+    runContainerResolverJob(javaProject, context);
   }
   
-  private static void runContainerResolverJob(IJavaProject javaProject) {
-    IEclipseContext context = EclipseContextFactory.getServiceContext(
-        FrameworkUtil.getBundle(CreateAppEngineStandardWtpProject.class).getBundleContext());
+  private static void runContainerResolverJob(IJavaProject javaProject, IEclipseContext context) {
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/BuildPath.java]: @chanseokoh's right: we don't need the class.  Just get the service context for this class' bundle.  I would revert this code to how it was.

The class/bundle is only important as it determines which bundle's lifecycle will be used to clean up any referenced OSGi services.

Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/BuildPath.java]: I can certainly do that. As currently written, this move is a fairly safe refactoring. However if I don't pass the class, then we're changing (I think) which bundle is used which does not seem to be a no-op. As long as that's OK, I can do it.

**Pull request merged**